src/core/bpf-cgroup-program-spec.c

```diff
@@ -40,3 +44,29 @@ int bpf_program_from_spec(struct BpfCgroupProgramSpec *spec, BPFProgram **ret) {
         *ret = TAKE_PTR(p);
         return 0;
 }
+
+int bpf_cgroup_program_spec_to_str(const struct BpfCgroupProgramSpec *spec, char *s, size_t sz) {
+        const char *atype, *aflags;
+        int r, pos = 0;
+
+        r = bpf_cgroup_attach_type_to_str(spec->attach_type, &atype);
+        if (r)
+                return r;
+
+        r = bpf_attach_flags_to_str(spec->attach_flags, &aflags);
+        if (r)
+                return r;
+
+        pos = snprintf(s, sz, "{ ");
+        if (!isempty(aflags))
+                pos += snprintf(s + pos, sz - pos,
+                                "\"attach_flags\" : \"%s\", ", aflags);
+
+        pos += snprintf(s + pos, sz - pos,
+                        "\"attach_type\" : \"%s\", "
+                        "\"bpffs_path\" : \"%s\" ", atype, spec->bpffs_path);
+
+        pos += snprintf(s + pos, sz - pos, "}");
+
+        return pos;
```

Comment:
this results in memory corruption if "sz" is too small...

Reply:
Removed JSON